lib/filewriter.js

```diff
@@ -138,16 +138,18 @@ FileWriter.prototype = {
             var guard, error = null;
             function finish(err) {
                 if (guard && global.clearTimeout) clearTimeout(guard);
+                var closeErr;
                 guard = null;
-                if (fd !== null) fs.close(fd, function(){});
+                if (fd !== null) try { fs.closeSync(fd) } catch (e) { closeErr = e }
                 fd = null;
-                return error ? cb(error) : cb(err);
+                return cb(error || err || closeErr);
             }
             try {
                 guard = setTimeout(function() {
-                    if (fd !== null) fs.close(fd, function(){});
+                    var closeErr;
+                    if (fd !== null) try { fs.closeSync(fd) } catch (e) { closeErr = e }
                     error = new Error("timed out waiting for last write");
-                    return finish(error);
+                    return finish(error || closeErr);
```

Comment:
The previous thread is gone because of the change, but your comment to my pointing out that since `error` is defined above, `closeErr` will never be returned here was `correct; the usage was intentional to emphasize that two different errors can be in play`.

I think if you want to catch and error and then ignore it, you should add a comment specifically mentioning that, and then just `return error`, rather than pretending that the return statement could potentially return one of the two, where in reality it cannot, which is misleading and looks like a logic error unless you're familiar with your rationale.